### DISCUSSION TITLE
How to link the grain boundary properly?

### DISCUSSION THREAD

**Discussion by MaynotbeGarychan** [mtex-toolbox/mtex#2166] [Ask Anything]
Hi,
Currently, I am making an interface to convert the grains object of MTEX to the geometry of grains and grain boundary in gmsh.
I know there are some outstanding tools, like MTEX2Gmsh, to achieve similar purpose.
But, I want to implement others features.
Now, I try to use the information of the grains.boundary to make the geometry. I noticed there are some useful midPoint, and direction of the line segments of the grains boundary. However, I don't know the sequence of this points. Therefore, when I convert them into Gmsh, the line segment is incorrect, see the attached.
Therefore, would you please tell me that when you are using "plot(grains.boundary)", how can you link them in a correct sequence?

Thanks!!
![Screenshot from 2024-07-09 15-01-24](https://github.com/mtex-toolbox/mtex/assets/59758494/a6674dce-ffad-455d-b4c9-3b877317eefe)
![Screenshot from 2024-07-09 14-59-59](https://github.com/mtex-toolbox/mtex/assets/59758494/eaf5e941-f7be-4e6b-9506-2ab40b0fc2b1)

**Comment by kilir** [comment]
Hi,

may have a look into `grainBoundary/plot`, there's a function called `plotOrdered2` as well as `plotSimple`, the former using `EulerCycles2'. I hope this helps.

Cheers,

Rüdiger
